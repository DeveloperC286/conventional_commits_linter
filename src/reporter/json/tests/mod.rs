use rstest::rstest;

use super::*;

#[rstest(
    message,
    snapshot_name,
    case(
        "feat()!: yargs-parser now throws on invalid combinations of config\n\n",
        "test_json_print_case_1"
    ),
    case("Release 1.0.126", "test_stdin_json_print_case_2"),
    case(
        "Merge branch 'fix-ocsp-signer-check' into 'master'",
        "test_json_print_case_3"
    )
)]
fn test_json_print(message: &str, snapshot_name: &str) {
    // Given
    let commit = Commit {
        oid: git2::Oid::zero(),
        message: message.to_string(),
    };
    let linting_errors = crate::linter::lint_commit(&commit, false);

    // When
    let reporting = print(message, &linting_errors);

    // Then
    insta::assert_snapshot!(snapshot_name, reporting);
}

#[rstest(
    messages,
    snapshot_name,
    case(
        &["feat()!: yargs-parser now throws on invalid combinations of config\n\n", "doc(webpack):webpack example (#1436)"],
        "test_json_print_all_case_1"
    ),
    case(
        &["refactor(ts support)!: ship yargs.d.ts (#1671)", "feat!: drop support for EOL Node 8 (#1686)"],
        "test_json_print_all_case_2"
    ),
)]
fn test_json_print_all(messages: &[&str], snapshot_name: &str) {
    // Given
    let commits: Vec<Commit> = messages
        .iter()
        .enumerate()
        .map(|(index, message)| Commit {
            oid: git2::Oid::from_str(&*format!("{}", index)).unwrap(),
            message: message.to_string(),
        })
        .collect();
    let linting_errors = crate::linter::lint_commits(&commits, false);

    // When
    let reporting = print_all(&commits, &linting_errors);

    // Then
    insta::assert_snapshot!(snapshot_name, reporting);
}
