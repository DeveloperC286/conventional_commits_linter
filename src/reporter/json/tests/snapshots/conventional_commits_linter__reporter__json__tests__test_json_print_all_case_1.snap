---
source: src/reporter/json/tests/mod.rs
expression: reporting

---
[{"commit_hash":"0000000000000000000000000000000000000000","commit_message":"feat()!: yargs-parser now throws on invalid combinations of config\n\n","linting_errors":["NonConventionalCommitsSpecification","EmptyScope"]},{"commit_hash":"1000000000000000000000000000000000000000","commit_message":"doc(webpack):webpack example (#1436)","linting_errors":["NonConventionalCommitsSpecification","NoSpaceAfterColonPrecedingTypeAndScope"]}]
